src/js/graphics.js

```diff
@@ -100,12 +120,18 @@ function regenText() {
 var editor_s_grille=[[0,1,1,1,0],[1,0,0,0,0],[0,1,1,1,0],[0,0,0,0,1],[0,1,1,1,0]];
 
 var spriteImages;
+
+function handleVector(name, vector) {
+    return vector.type === 'canvas' ? createJsonSprite(name, vector)
+    : undefined;
+}
+
```

Comment:
TODO: currently `undefined` is returned when type does not equal `canvas`, should probably done another way, e.g. by throwing an exception

Reply:
The logic is OK, but I prefer to return null. I use exceptions only for things I can't and shouldn't happen.